conans/client/api/conan_api.py

```diff
@@ -268,23 +269,100 @@ def _get_revisions(self, ref, remote=None):
+        """
+        Get all the recipe revisions given a reference from cache or remote.
+
+        :param reference: `ConanFileReference` without the revision
+        :param remote: `Remote` object
+        :return: `dict` with all the results, e.g.,
+                 {"remote": "my_remote_name",  # or None
+                  "reference": "libyaml/0.2.5",
+                  "results": [{"revision": "80b7cbe095ac7f38844b6511e69e453a",
+                               "time": "2021-07-20 00:56:25 UTC"}, ...]}
+        """
         ref = ConanFileReference.loads(reference)
         if ref.revision:
             raise ConanException(f"Cannot list the revisions of a specific recipe revision")
 
-        return self._get_revisions(ref, remote=remote)
+        results = self._get_revisions(ref, remote=remote)
+        return {"remote": remote.name if remote else None,
+                "reference": reference,
+                "results": results}
+
+    @api_method
+    def get_package_ids(self, reference, remote=None):
+        """
+        Get all the Package IDs given a recipe revision from cache or remote.
+
+        Note: if reference does not have the revision, we'll return the Package IDs for
+        the latest recipe revision by default
+
+        :param reference: `ConanFileReference` with/without revision
+        :param remote: `Remote` object
+        :return: `list` of `dict` with the `package-id` as keys, e.g.,
+                 {"remote": "my_remote_name",  # or None
+                  "reference": "libcurl/7.77.0#2a9c4fcc8d76d891e4db529efbe24242",
+                  "results": {"d5f16437dd4989cc688211b95c24525589acaafd": {
+                                    "settings": {"compiler": "apple-clang",...},
+                                    "options": {'options': {'shared': 'False',...}},
+                                    "requires": [
+                                        'mylib/1.0.8:3df6ebb8a308d309e882b21988fd9ea103560e16',...]}
+                            }
+                 }
+        """
+        ref = ConanFileReference.loads(reference)
+        if remote:
+            rrev = ref if ref.revision else self.app.remote_manager.get_latest_recipe_revision(ref,
+                                                                                               remote)
+            try:
+                packages_props = self.app.remote_manager.search_packages(remote, rrev, None)
+            except NotFoundException:
+                # This exception must be catched manually due to a server inconsistency:
+                # Artifactory API returns an empty result if the recipe doesn't exist, but
+                # Conan Server returns a 404. This probably should be fixed server side,
+                # but in the meantime we must handle it here
+                packages_props = {}
+        else:
+            rrev = ref if ref.revision else self.app.cache.get_latest_rrev(remote, ref)
```

Comment:
This function does not have the remote argument...
I think we are missing some tests, check how this one fails:
```
    def test_existing_reference(self):
        self.client.save({"conanfile.py": GenConanfile().with_name("pkg").with_version("1.0")})
        self.client.run(f"create .")
        self.client.run(f"list package-ids pkg/1.0")
```

Reply:
Completely right. The thing is that I have a test for cache and remote cases but as you said I'm missing something... Thanks for the catch @czoido